plugins/modules/ndo_endpoint_mac_tag_policy.py

```diff
@@ -0,0 +1,497 @@
+    mac = module.params.get("endpoint_mac_address")
+    uuid = module.params.get("uuid")
+    bd = module.params.get("bridge_domain")
+    bd_uuid = module.params.get("bridge_domain_uuid")
+    vrf = module.params.get("vrf")
+    vrf_uuid = module.params.get("vrf_uuid")
+    annotations = module.params.get("annotations")
+    policy_tags = module.params.get("policy_tags")
+    state = module.params.get("state")
+    reference_dict = {
+        "bd": {
+            "name": "bdName",
+            "reference": "bdRef",
+            "type": "bd",
+            "template": "bdTemplateName",
+            "templateId": "bdTemplateId",
+            "schema": "bdSchemaName",
+            "schemaId": "bdSchemaId",
+        },
+        "vrf": {
+            "name": "vrfName",
+            "reference": "vrfRef",
+            "type": "vrf",
+            "template": "vrfTemplateName",
+            "templateId": "vrfTemplateId",
+            "schema": "vrfSchemaName",
+            "schemaId": "vrfSchemaId",
+        },
+    }
+
+    if mac and not uuid and not (bd or vrf or vrf_uuid or bd_uuid):
+        mso.fail_json(msg="when providing a MAC address without UUID, one of the following is required: bridge_domain, bridge_domain_uuid, vrf, vrf_uuid")
+
+    mso_template = mso_templates.get_template("tenant", template_name, template_id)
+    mso_template.validate_template("tenantPolicy")
+
+    tenant_id = mso_template.template.get("tenantPolicyTemplate", {}).get("template", {}).get("tenantId")
+    templates_objects_path = "templates/objects"
+    if vrf and not vrf_uuid:
+        vrf_object = mso_template.get_vrf_object(vrf, tenant_id, templates_objects_path)
+        vrf_uuid = vrf_object.details.get("uuid")
+    elif bd and not bd_uuid:
+        bd_object = mso_template.get_bd_object(bd, tenant_id, templates_objects_path)
+        bd_uuid = bd_object.details.get("uuid")
+
+    match = mso_template.get_endpoint_mac_tag_policy_object(uuid, mac, bd_uuid, vrf_uuid)
+
+    if (uuid or (mac and (bd_uuid or vrf_uuid))) and match:
+        mso.existing = mso.previous = copy.deepcopy(
+            mso_template.update_config_with_template_and_references(match.details, reference_dict)
+        )  # Query a specific object
+    elif match:
+        mso.existing = [mso_template.update_config_with_template_and_references(obj, reference_dict) for obj in match]  # Query all objects
+
+    endpoint_mac_tag_policy_path = "/tenantPolicyTemplate/template/endpointMacTagPolicies/{0}".format(match.index if match else "-")
+
+    ops = []
+
+    if state == "present":
+
+        mso_values = {
+            "mac": mac,
+            "bdRef": bd_uuid,
+            "vrfRef": vrf_uuid,
+        }
+
+        if annotations:
+            mso_values["tagAnnotations"] = [
+                {
+                    "tagKey": annotation.get("key"),
+                    "tagValue": annotation.get("value"),
+                }
+                for annotation in annotations
+            ]
```

Comment:
shall we create a function for this, since it is also used in ip? perhaps in time also in other modules?